scripts/fix_lotus_encumbrances/fix_lotus_encumbrances.py

```diff
@@ -496,22 +511,97 @@ async def release_unreleased_encumbrances_for_closed_orders(closed_orders_ids, f
-        ids = []
-        for fund in funds:
-            ids.append(fund.get('id'))
     except Exception as err:
         print('Error getting order ids with query "{}": {}'.format(query, err))
         raise SystemExit(1)
-    return ids
+    return funds
+
+
+async def get_polines_by_order_id(order_id):
+    query = 'purchaseOrderId=={}'.format(order_id)
+    resp = await get_request(okapi_url + 'orders-storage/po-lines', query)
+    po_lines = resp['poLines']
+    return po_lines
+
+
+def lookup_fund(poline_id, code, all_funds):
+    for fund in all_funds:
+        if fund['code'] == code:
+            return fund['id']
+    print('Fund code {} from poline {} not exists. Correct the data and rerun the script'.format(code, poline_id))
+    raise SystemExit(1)
+
+
+async def process_order_encumbrances_relations(order_id, fiscal_year_id, all_funds, order_sem):
+    po_lines = await get_polines_by_order_id(order_id)
+
+    poline_futures = []
+    if len(po_lines) > 0:
+        for po_line in po_lines:
+            poline_futures.append(asyncio.ensure_future(process_po_line_encumbrances_relations(po_line, fiscal_year_id, all_funds)))
+        await asyncio.gather(*poline_futures)
+
+    order_sem.release()
+
+
+async def check_if_fd_needs_updates(poline_id, fd, query):
+    encumbrances = await get_encumbrances_by_query(query)
+    if len(encumbrances) > 0 and encumbrances[0]['id'] != fd['encumbrance']:
+        print('Updating poline {} encumbrance {} with new value {}'.format(poline_id, fd['encumbrance'], encumbrances[0]['id']))
+        fd['encumbrance'] = encumbrances[0]['id']
+        return True
+    return False
+
+
+async def process_po_line_encumbrances_relations(poline, fiscal_year_id, all_funds):
+    enc_futures = []
+    for fd in poline['fundDistribution']:
+        if fd.get('encumbrance') is not None:
+            query = 'amount<>0.0 AND encumbrance.sourcePoLineId=={} AND fiscalYearId<>{} AND fromFundId=={}'.format(
+                poline['purchaseOrderId'], fiscal_year_id, fd.get('fundId'))
+            enc_futures.append(asyncio.ensure_future(check_if_fd_needs_updates(poline['id'], fd, query)))
+    poline_needs_updates = await asyncio.gather(*enc_futures)
+
+    # update poline if one or more fund distributions modified
+    if True in poline_needs_updates:
+        url = '{}orders-storage/po-lines/{}'.format(okapi_url, poline.get('id'))
+        await put_request(url, poline)
+
+
+# Get po_lines by order id
+    # for each poline process fund distributions
+        # for each fund distribution check encumbrance relationship and modify if needed -
+        # in case if encumbrance id specified in fund distribution:
+        # get encumbrance by poline id and current FY<>transaction.FY and amount <> 0
+        # if fd.encumbrance != transaction.id --> set new encumbrance reference
+        # update poline if modified
+async def fix_poline_encumbrances_relations(open_orders_ids, fiscal_year_id, all_funds):
```

Comment:
please update readme with this new step:
https://github.com/folio-org/mod-finance-storage/blob/master/scripts/fix_lotus_encumbrances/README.md

Reply:
updated readme file